homework-online-store/js/main.js

```diff
@@ -0,0 +1,164 @@
+  } else if (stock == 1) {
+    stockEl.textContent = 'last chance';
+    stockEl.style.color = 'orange';
+  } else if (stock == 0) {
+    stockEl.textContent = 'out of stock';
+    stockEl.style.color = 'red';
+  }
+  return stockEl;
+}
+
+function makeProductCardInCart(product) {
+  const productCardEl = document.createElement('div');
+  productCardEl.classList.add('js-cart__list__item');
+  productCardEl.innerHTML = `
+    <hr>
+    <img src="${product.extra_data.image}" alt="${product.title}">
+    <div class="js-cart__list__info">
+      <h4>${product.extra_data.brand} ${product.extra_data.type}</h4>
+      <span>${product.title}</span><br>
+      <span>Price: $ ${product.price}</span>
+    </div>
+  `; // <button class="js-cart__list__item__delete">delete</button>
+  return productCardEl;
+}
+
+function displayProduct(product) {
+  const productItemEl = document.createElement('li');
+  productItemEl.innerHTML = makeProductCard(product);
+  goodsListEl.appendChild(productItemEl);
+  const addToCartBtnEl = productItemEl.querySelector('.js-product-card__cart-button');
+  addToCartBtnEl.addEventListener('click', () => {
+    if (addToCartBtnEl.classList.contains('plus')) {
+      goodsInCart.push(product);
+      amount += product.price;
+    } else if (addToCartBtnEl.classList.contains('minus')) {
+      goodsInCart.splice(goodsInCart.indexOf(product), 1);
+      amount -= product.price;
+    }
+    badgeEl.textContent = goodsInCart.length;
+    addToCartBtnEl.classList.toggle('plus');
+    addToCartBtnEl.classList.toggle('minus');
+  });
+}
+
+function toggleVisible(element) {
+  element.classList.toggle('hidden');
+}
+
+function renderPage(goods) {
+  goods.forEach(displayProduct);
+  cartSnippetEl.addEventListener('click', event => {
+    event.preventDefault();
+    if (goodsInCart.length == 0) return;
+    cartListEl.innerHTML = '';
+    goodsInCart.forEach(good => {
+      cartListEl.appendChild(makeProductCardInCart(good));
+      // const deleteBtnEl = cartEl.querySelector('.js-cart__list__item__delete');
+      // deleteBtnEl.addEventListener('click', event => {
+      //   console.log(event.target.parentNode);
+      //   const delEl = event.target.parentNode;
+      //   cartListEl.removeChild(delEl);
+      // });
+    });
+    cartAmountEl.textContent = goodsInCart.reduce((total, product) => {
+      return total + product.price;
+    }, 0);
+    toggleVisible(cartEl);
+  });
+  closeCartEl.addEventListener('click', () => toggleVisible(cartEl));
+}
+
+function renderStock(good) {
+  const goodItemEl = document.querySelector(`.${good.sku}`);
+  goodItemEl.querySelector('.js-product-card__cart-button').disabled = isNull(good.stock);
+  goodItemEl.querySelector('.js-product-card__brand').nextElementSibling = checkStock(good.stock).outerHTML;
+}
+
+async function deleteFromStock(token, sku) {
+  return await fetch(`https://student-store.travisshears.xyz/buy/${sku}?token=${token}`, {
+    method: 'GET'
```

Comment:
So the problem is here. This must be a `POST` method not a `GET`. 🤦 I wrote the homework wrong... 

How did I debug this? Well the line 149 `.then(response => response.json())` was failing so I commended it out, that did not seem to help. Thought maybe the endpoint was not returning json but regular string so I tied `.then(response => response.text())` 

that printed:
```<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>405 Method Not Allowed</title>
<h1>Method Not Allowed</h1>
<p>The method is not allowed for the requested URL.</p>
```

Progress! 👍 seems the API endpoint exist but is not setup to receive get request. more here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

So I thought hmmm maybe a POST request instead.. and that worked.

Reply:
I've since updated the homework fixing this mistake. Sorry again. Good work . https://mrtim.github.io/js-munich-2020-spring/lessons/10_conditions/classwork_advanced

Reply:
thank you for describing of debugging process! I was trying to do that way, also saw 405 Error. But I couldn't imagine that I shouldn't trust your instruction with GET request 😄

I've merged this pull request (accidentally, through another branch). But tomorrow I open a new one, with new questions I guess 🙄